pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -322,18 +326,175 @@ downloadPhase:
 	ppln.phaseWaiter.Done()
 }
 
+// extractClusterScopedAppOnPod untars the given app package to the bundle push location
+func extractClusterScopedAppOnPod(worker *PipelineWorker, appSrcScope string, appPkgPathOnPod, appPkgLocalPath string) error {
+	cr := worker.cr
+	scopedLog := log.WithName("extractClusterScopedAppOnPod").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace(), "app name", worker.appDeployInfo.AppName)
+
+	if appSrcScope != enterpriseApi.ScopeCluster {
+		//Nothing to be done in case of local scoped apps
+		scopedLog.Error(nil, "app extraction is called for non-cluster scoped apps", "scope", appSrcScope)
+		return nil
+	}
+
+	var clusterAppsPath string
+	kind := worker.cr.GroupVersionKind().Kind
+	if kind == "SearchHeadCluster" {
+		clusterAppsPath = "/opt/splunk/etc/shcluster/apps/"
+	} else if kind == "ClusterMaster" {
+		clusterAppsPath = "/opt/splunk/etc/master-apps/"
+	} else {
+		// Do not return an error
+		scopedLog.Error(nil, "app extraction should not be called", "kind", kind)
+		return nil
+	}
+
+	// untar the package to the cluster apps location, then delete it
+	// ToDo: sgontla: cd, tar, and rm commands are trivial commands. packign together to avoid spanning multiple processes.
+	// A better alternative is to maintain a script (that can give us the status of each command that we can map into a logical error, and copy if when needed.). Alternatively, we can mount it through a configMap
+	command := fmt.Sprintf("cd %s; tar -xzf %s; rm -rf %s", clusterAppsPath, appPkgPathOnPod, appPkgPathOnPod)
+	streamOptions := &remotecommand.StreamOptions{
+		Stdin: strings.NewReader(command),
+	}
+
+	stdOut, stdErr, err := splutil.PodExecCommand(*worker.client, worker.targetPodName, cr.GetNamespace(), []string{"/bin/sh"}, streamOptions, false, false)
+	if err != nil {
+		scopedLog.Error(err, "app package untar & delete failed", "stdout", stdOut, "stderr", stdErr)
+		return err
+	}
+
+	// Now that the App package was moved to the persistent location on the Pod.
+	// Remove the app package from the Operator storage area
+	// Note:- local scoped app packages are removed once the installation is complete(for all the replica members)
+	err = os.Remove(appPkgLocalPath)
+	if err != nil {
+		// Issue is local, so just log an error msg and do *NOT* return an error here
+		scopedLog.Error(err, "failed to delete", "app pkg", appPkgLocalPath)
+	}
+
+	return nil
+}
+
+// runPodCopyWorker runs one pod copy worker
+func runPodCopyWorker(worker *PipelineWorker, ch chan struct{}) {
+	cr := worker.cr
+	scopedLog := log.WithName("runPodCopyWorker").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace(), "app name", worker.appDeployInfo.AppName)
+	defer func() {
+		<-ch
+		worker.isActive = false
+		worker.waiter.Done()
+	}()
+
+	appPkgFileName := worker.appDeployInfo.AppName + "_" + strings.Trim(worker.appDeployInfo.ObjectHash, "\"")
```

Comment:
Ya, i saw that, But I need just the file name alone so that I can use the same for both local and remote Pod. Probably, we can  rename `getLocalAppFileName ` to `getLocalAppFilePath`, then create a new one `getLocalAppFileName` that just gives the app pkg name.